### PR TITLE
Exclude libraries from auditwheel

### PR DESCRIPTION
**Pull request by SimonRit**
This PR
- greatly simplifies the handling of the --exclude_libs option thanks to a new --exclude option of auditwheel. This has been tested and seems to work, see [CudaCommon CI packaging](https://github.com/RTKConsortium/ITKCudaCommon/actions/runs/7904605753) and the result (compared to #265):
```
- srit@srit-Precision-5480:~/tmp$ unzip LinuxWheel310-cuda116.zip 
Archive:  ../LinuxWheel310-cuda116.zip
  inflating: itk_cudacommon_cuda116-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl  
  inflating: itk_cudacommon_cuda116-1.0.1-cp310-cp310-manylinux_2_28_x86_64.whl  
srit@srit-Precision-5480:~/tmp$ unzip itk_cudacommon_cuda116-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
Archive:  itk_cudacommon_cuda116-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
   creating: itk/
   creating: itk_cudacommon_cuda116-1.0.1.dist-info/
   creating: itk_cudacommon_cuda116.libs/
   creating: itk/Configuration/
  inflating: itk/CudaCommonPython.py  
  inflating: itk/_CudaCommonPython.so  
  inflating: itk/itkCudaDataManagerPython.py  
  inflating: itk/itkCudaImageDataManagerPython.py  
  inflating: itk/itkCudaImagePython.py  
  inflating: itk/itkImageSourceCudaCommonPython.py  
  inflating: itk/itkImageToImageFilterCudaCommonPython.py  
  inflating: itk/Configuration/CudaCommonConfig.py  
  inflating: itk/Configuration/CudaCommon_snake_case.py  
  inflating: itk_cudacommon_cuda116-1.0.1.dist-info/LICENSE  
  inflating: itk_cudacommon_cuda116-1.0.1.dist-info/METADATA  
  inflating: itk_cudacommon_cuda116-1.0.1.dist-info/WHEEL  
  inflating: itk_cudacommon_cuda116-1.0.1.dist-info/top_level.txt  
  inflating: itk_cudacommon_cuda116-1.0.1.dist-info/RECORD  
  inflating: itk_cudacommon_cuda116.libs/libtbb-b694c9f9.so.12.9  
```
- includes in the docker LD_LIBRARY_PATH the library paths to forward it to auditwheel.

### CODE REVIEW

**Comment by SimonRit** [comment]
cc @LucasGandel

File: scripts/dockcross-manylinux-build-module-wheels.sh [hunk 1]
```diff
@@ -61,12 +61,18 @@ chmod 777 $(pwd)/tools
 mkdir -p dist
 DOCKER_ARGS="-v $(pwd)/dist:/work/dist/ -v ${script_dir}/..:/ITKPythonPackage -v $(pwd)/tools:/tools"
 DOCKER_ARGS+=" -e MANYLINUX_VERSION"
+DOCKER_ARGS+=" -e LD_LIBRARY_PATH"
 # Mount any shared libraries
 if [[ -n ${LD_LIBRARY_PATH} ]]; then
   for libpath in ${LD_LIBRARY_PATH//:/ }; do
-	  DOCKER_ARGS+=" -v ${libpath}:/usr/lib64/$(basename -- ${libpath})"
+          DOCKER_LIBRARY_PATH="/usr/lib64/$(basename -- ${libpath})"
+	  DOCKER_ARGS+=" -v ${libpath}:${DOCKER_LIBRARY_PATH}"
```
Comment (dzenanz) [inline scripts/dockcross-manylinux-build-module-wheels.sh:69]: Should we use spaces consistently? One tab character sneaked in here.

**Review by dzenanz — APPROVED** [review]
This looks like a great fix.

File: scripts/dockcross-manylinux-build-module-wheels.sh (hunk 1 above)
Comment (SimonRit) [reply scripts/dockcross-manylinux-build-module-wheels.sh]: Sorry, should be fixed now.

**Review by thewtex — APPROVED** [review]
@SimonRit thanks!

LGTM

@LucasGandel please take a look

**Review by LucasGandel — APPROVED** [review]

**Comment by SimonRit** [comment]
> I can't tell for the LD_LIBRARY_PATH change though.

I made it because auditwheel could not find the shared libraries otherwise. Maybe it was working before thanks to the `whitelist`? I honestly don't know why...

**Comment by dzenanz** [comment]
How does [this](https://github.com/RTKConsortium/ITKCudaCommon/pull/33/commits/bdc9b37b06fb3831c4ca833ea249a9ed912831df#diff-b605ee46fbd21ee607667ab55c9a27d46cc6981d9468c2db4bed457d80aa9ac3R8-R9) change in light of this PR being merged? I tried using it [here](https://github.com/dzenanz/ANTsWasm/commit/369b9b5f5e8d5a6df6f14cc57493ffb84b9a863f), without [much luck](https://github.com/dzenanz/ANTsWasm/actions/runs/7922130975/job/21629020185).

**Comment by SimonRit** [comment]
> without [much luck](https://github.com/dzenanz/ANTsWasm/actions/runs/7922130975/job/21629020185).

The error message in the failing build seems unrelated
`Could NOT find SWIG (missing: SWIG_DIR)`
I'm not sure what's the problem here...

**Comment by dzenanz** [comment]
Maybe it is related to 96e34770ba4d4a989eb235afc15e68439c5a94c3?

**Pull request merged**